Core/src/Geometry/CylinderVolumeStack.cpp

```diff
@@ -0,0 +1,1020 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Geometry/CylinderVolumeStack.hpp"
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Definitions/Common.hpp"
+#include "Acts/Definitions/Tolerance.hpp"
+#include "Acts/Geometry/CylinderVolumeBounds.hpp"
+#include "Acts/Utilities/BinningType.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <memory>
+#include <sstream>
+
+namespace Acts {
+
+struct CylinderVolumeStack::VolumeTuple {
+  Volume* volume;
+  const CylinderVolumeBounds* bounds{};
+  std::shared_ptr<CylinderVolumeBounds> updatedBounds{};
+  Transform3 localTransform;
+  Transform3 globalTransform;
```

Comment:
```suggestion
  Transform3 localTransform = Transform3::Identity();
  Transform3 globalTransform = Transform3::Identity();
```
(didn't check if this compiles 😬 )